File: web/src/main/java/me/robinvdb/runetracker/database/RunescapeUserDatabase.java
package me.robinvdb.runetracker.database;

import me.robinvdb.runetracker.domain.RunescapeUser;

import java.util.Collection;
import java.util.Optional;
import java.util.stream.Stream;

public interface RunescapeUserDatabase {
    void add(RunescapeUser user);

    void addAll(Collection<RunescapeUser> users);

    void delete(RunescapeUser user);

    Optional<RunescapeUser> get(int id);

    Stream<RunescapeUser> all();

    void update(RunescapeUser user);
}

File: web/src/main/java/me/robinvdb/runetracker/database/RunescapeUserDatabase.kt
package me.robinvdb.runetracker.database

import me.robinvdb.runetracker.domain.RunescapeUser
import java.util.Optional
import java.util.stream.Stream

interface RunescapeUserDatabase {
    fun add(user: RunescapeUser)

    fun addAll(users: Collection<RunescapeUser>)

    fun delete(user: RunescapeUser)

    operator fun get(id: Int): Optional<RunescapeUser>

    fun all(): Stream<RunescapeUser>

    fun update(user: RunescapeUser)
}
